hathor/manager.py

```diff
@@ -1173,6 +1177,15 @@ def get_cmd_path(self) -> Optional[str]:
         """Return the cmd path. If no cmd path is set, returns None."""
         return self._cmd_path
 
+    def update_peer_id(self) -> None:
+        """Update this manager's PeerId from the json file. Only entrypoints can be changed."""
+        if not self._peer_id_json_path:
+            return
+
+        new_peer_id = PeerId.create_from_json_path(self._peer_id_json_path)
+        assert new_peer_id.id == self.my_peer.id, 'The Peer ID cannot be changed.'
```

Comment:
Since this comes from user data that the node loads, I think it should not be an assert. It could be an error log that says the new `peer.json` file was ignored because the peer_id doesn't match, or something like that. But without raising any exceptions.

Reply:
Agreed, changed in https://github.com/HathorNetwork/hathor-core/pull/981/commits/cccfe886117aea2037a5c831fc23d8f1eeb14ab7